src/fs_core.ts

```diff
@@ -59,14 +73,17 @@ export class SyncOPFSFile {
     return new wasi.Filestat(wasi.FILETYPE_REGULAR_FILE, this.size);
   }
 
-  truncate() {
-    return this.handle.truncate(0);
+  truncate(): number {
+    if (this.readonly) return wasi.ERRNO_PERM;
+    this.handle.truncate(0);
+    return wasi.ERRNO_SUCCESS;
   }
 
 }
 
 export class Directory {
   contents: { [key: string]: File | Directory | SyncOPFSFile };
+  readonly = false; // FIXME implement, like marking all files within readonly?
```

Comment:
Not sure what to do here. On Linux you can e.g. have an `r-x` directory, which disallows creation of new files but othermise might contain writable files.

Reply:
I think the readonly flag for directories should be left out at least until full unix permissions are implemented.